src/bluecore/app/change_notifications/activity_streams.py

```diff
@@ -0,0 +1,349 @@
+from bluecore.schemas import (
+    ActivityStreamsChangeSetSchema,
+    ActivityStreamsEntityChangeActivitiesSchema,
+    ActivityStreamsEntryPointSchema,
+    ActivityStreamsObjectSchema,
+)
+from bluecore.utils.constants import ACTIVITY_STREAMS_PAGE_LENGTH, BCType, BFType
+from bluecore_models.models import ResourceBase, Version
+from sqlalchemy import func
+from sqlalchemy.orm import Session
+from typing import Dict, List, Union
+import math
+import os
+
+HOST = os.getenv("HOST", "http://127.0.0.1:3000")
+
+
+class ActivityStreamsGenerator:
+    """
+    A class to handle change notifications.
+    """
+
+    page_length: int = 1
+
+    def __init__(self, page_length: int = ACTIVITY_STREAMS_PAGE_LENGTH):
+        """
+        Initializes the ActivityStreamsGenerator with a specified page length.
+        The page length is set to ACTIVITY_STREAMS_PAGE_LENGTH (100) by default.
+        Don't change this value unless running tests.
+
+        Args:
+            page_length (int): The number of items per page for the activity streams.
+        """
+        self.page_length = page_length
+
+    def activity_streams_feed(
+        self, db: Session, bc_type: str
+    ) -> ActivityStreamsEntryPointSchema:
+        """
+        Generates an activity streams feed for a given resource type.
+
+        This method constructs an OrderedCollection representation of an
+        activity streams feed for the specified resource type. It calculates
+        the total number of items, determines the last page, and provides
+        metadata about the feed including its context, summary, and pagination
+        details.
+        The feed is structured according to the Activity Streams 2.0
+        specification, and includes a context URL for the EMM specification.
+
+        Args:
+            db (Session): database session used to query the resource data
+            bf_type (str): type of the resource for which the activity
+                  streams feed is being generated
+
+        Returns:
+            ActivityStreamsEntryPointSchema: activity streams feed
+                  in the OrderedCollection format, including metadata
+                  and pagination details
+        """
+        total = (
+            db.query(func.count(Version.id))
+            .join(ResourceBase)
+            .filter(ResourceBase.type == bc_type)
+            .scalar()
+        )
+        last_page: int = math.ceil(total / self.page_length)
+        return ActivityStreamsEntryPointSchema(
+            context=[
+                "https://www.w3.org/ns/activitystreams",
+                "https://emm-spec.org/1.0/context.json",
+            ],
+            summary="Bluecore",
+            type="OrderedCollection",
+            id=f"{HOST}/change_documents/{bc_type}/activitystreams/feed",
```

Comment:
Should be refactored to `f"{self.host}/change_documents/{bc_type}/activitystreams/feed`